lib/containers/engine.pm

```diff
@@ -137,8 +145,13 @@ sub run_container {
     my $params = sprintf qq(%s %s %s), $keep_container, $mode, $name;
     my $cmd = sprintf qq(run %s %s %s), $params, $image_name, $remote;
     record_info "cmd_info", "Container executes:\noptions $params $image_name $remote";
-    my $retries = $args{retry} // 1;
-    return $self->_engine_script_retry($cmd, timeout => $args{timeout}, retry => $retries, delay => $args{delay});
+    if ($args{validate}) {
+        die "retry and validate argument cannot be given at the same time" if ($args{retry});
```

Comment:
why `die` and not continue with `validation` and print a warning, so we can fix it misused invocations later.

Reply:
Using `validate` and `retry` here is incorrect usage of the function and should by definition die.

I favor the approach that a function should fail early rather than have unexpected functionality. Here, if the user passes `validate` and `retry` (which is stated as not supported) the function will fail.
Your suggestion would silently ignore the `retry` argument, so someone could (and will) wrongly assumes that it would work otherwise.

Let's save ourselves some gray hair.

Reply:
I don't want to block this, but just that you know the `script_retry()` actually can get you the output with `die => 0`, e.g.:
```
my $ret = script_retry($cmd, retry => 3, delay => 60, timeout 30, die => 0);
```

Reply:
`script_retry` returns the return code of `$cmd` but not the output - See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/utils.pm#L1537

Reply:
I'm not sure why retry will not be used with validate_output... Maybe you could use the retry approach from here: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/wicked/wlan.pm#L324

Reply:
In that case we would have a functionality implemented, which is not used or tested anywhere. We could implement it now, but then we would also do add some unit tests or use it somewhere, just to make sure that this case is also functional.

At this point a failure seems to be more economical. We can implement this at any time we need it but we don't have to invest time and effort now into a functionality, which probably is never being used.

Reply:
I mark this as resolved, as the original approach is not handled anymore.